dht/i_partitioner.cc

```diff
@@ -36,24 +36,37 @@
 
 namespace dht {
 
-i_partitioner::i_partitioner(unsigned shard_count, unsigned sharding_ignore_msb_bits)
+sharding_info::sharding_info(unsigned shard_count, unsigned sharding_ignore_msb_bits)
     : _shard_count(shard_count)
     // if one shard, ignore sharding_ignore_msb_bits as they will just cause needless
     // range breaks
     , _sharding_ignore_msb_bits(shard_count > 1 ? sharding_ignore_msb_bits : 0)
     , _shard_start(init_zero_based_shard_start(_shard_count, _sharding_ignore_msb_bits))
 {}
 
+i_partitioner::i_partitioner(unsigned shard_count, unsigned sharding_ignore_msb_bits)
+    : _sharding_info(shard_count, sharding_ignore_msb_bits)
+{}
+
 unsigned
-i_partitioner::shard_of(const token& t) const {
+sharding_info::shard_of(const token& t) const {
```

Comment:
As I said in a top-level comment, this is more consistently named "sharded", but we can delay renaming until after the dust settles.